core/lib/instrument_api.h

```diff
@@ -1763,6 +1763,17 @@ DR_API
 const char *
 dr_get_application_name(void);
 
+DR_API
+/**
+ * Sets the passed arguments to the command-line arguments of the
+ * application.
+ * @param argc Pointer to place argc
+ * @param argv Pointer to place argv
```

Comment:
hmm, following the above comment, we also need to agree a bit about the final API function prototype. 

I would like to have the address that _refers_ to the memory containing the args. It is for this reason why my PR has the API function in the form: 

`dr_get_application_cl_args(OUT int **argc, OUT char ***argv); `

For example, this would be useful for clients that perform some memory checking analysis. In fact, the function enables me to mark args as source for data-flow analysis. For windows (which I do not consider atm), the interface suggested won't be suitable.

Reply:
For that it sounds like a "memory blob" interface akin to drsyscall's for syscall int/out params is what comes to mind.  Would this move more in the direction of "memory blobs of data set up for a new process" which would include environment variables, auxv, etc.?  Though it also seems useful to get the app command line.  Maybe they should be separated.

Reply:
The only difference between a blob vs command-line api interface would the that the latter provides parsing right? Can't we let the user do the parsing?

Reply:
A blob interface might merge argc, argv, env, and auxv into a single contiguous memory region of "input data" with no semantic labels as to what is what.  I don't think the user would be able to parse that.

Reply:
Ah okay, I was thought you meant that the blob would only consist of argv data.